src/ast-utils.ts

```diff
@@ -50,3 +52,35 @@ export function siblingsAfter(node: ASTNode): ASTNode[] {
   const siblings: ASTNode[] = Array.from(node.parentNode!.childNodes!);
   return siblings.slice(siblings.indexOf(node) + 1);
 }
+
+/**
+ * The results of `queryAll` combined with `querySelectorAllTemplates`.
+ */
+export function querySelectorAllWithTemplates(
+    node: ASTNode, predicate: dom5.Predicate): ASTNode[] {
+  const results = dom5.queryAll(node, predicate);
+  const templates = dom5.queryAll(node, matchers.template);
+  for (const template of templates) {
+    const content = treeAdapters.default.getTemplateContent(template);
+    for (const match of dom5.queryAll(content, predicate)) {
+      results.push(match);
+    }
+  }
+  return results;
+}
+
+/**
+ * Find content inside <template> tags.
+ */
+export function querySelectorAllTemplates(
+    node: ASTNode, predicate: dom5.Predicate): ASTNode[] {
+  const results: ASTNode[] = [];
+  const templates = dom5.queryAll(node, matchers.template);
+  for (const template of templates) {
+    const content = treeAdapters.default.getTemplateContent(template);
```

Comment:
What @usergenic said. I can get a typings PR submitted, but using this API seems like the path of least resistance for now.